crates/sui-core/src/authority.rs

```diff
@@ -1699,86 +1699,14 @@ impl AuthorityState {
 
 #[async_trait]
 impl ExecutionState for AuthorityState {
-    type Transaction = ConsensusTransaction;
     type Error = SuiError;
-    type Outcome = Vec<u8>;
-
-    /// This function will be called by Narwhal, after Narwhal sequenced this certificate.
-    #[instrument(level = "trace", skip_all)]
-    async fn handle_consensus_transaction(
```

Comment:
It seems we kind of need this code, why are we removing it?

Reply:
this is no longer part of `ExecutionState` but instead of `BatchExecutionState `

Reply:
Batch execution state is used by executors that wish to handle the entire batch by themselves. I would not break this abstraction unless we really need to because other people depend on it (eg. protocol lab)

Reply:
actually where do you see this code in the PR (it seems simply gone)?

Reply:
to clarify, what trait do we want AuthorityState to implement?